Transpiler integration refactor and qiskit v1.2 support (#116)
iqm-finland/qiskit-on-iqm: docs/user_guide.rst

::

    global phase: 3π/2
              ┌─────────────┐                  ┌─────────────┐ ░       ┌─┐
    q_2 -> 5  ┤ R(π/2,3π/2) ├──────────■───────┤ R(π/2,5π/2) ├─░───────┤M├
              ├─────────────┤          │       └─────────────┘ ░ ┌─┐   └╥┘
    q_0 -> 10 ┤ R(π/2,3π/2) ├─■────────■───────────────────────░─┤M├────╫─
              ├─────────────┤ │ ┌─────────────┐                ░ └╥┘┌─┐ ║ 
    q_1 -> 15 ┤ R(π/2,3π/2) ├─■─┤ R(π/2,5π/2) ├────────────────░──╫─┤M├─╫─
              └─────────────┘   └─────────────┘                ░  ║ └╥┘ ║ 
      meas: 3/════════════════════════════════════════════════════╩══╩══╩═
                                                                0  1  2 

Under the hood the Qiskit transpiler uses the :class:`.IQMDefaultSchedulingPlugin` plugin that automatically adapts the
transpiled circuit from Qiskit to the IQM backend. In particular, if the `optimization_level >= 0`, the plugin will use
the :class:`.IQMOptimizeSingleQubitGates` pass to optimize single-qubit gates, and the :class:`.IQMNaiveResonatorMoving`

> to insert :class:`.MoveGate` instructions for devices that have a support resonators. 

Reviewer: ```suggestion
to insert :class:`.MoveGate` instructions for devices whose physical transpiler target has resonators.
```
